src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const tempState = { ...state };
+  const statesArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(tempState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const rem of action.keysToRemove) {
+          delete tempState[rem];
+        }
+        break;
+
+      case 'clear': {
+        for (const key in tempState) {
+          delete tempState[key];
+        }
+        break;
+      }
+
+      default:
+        break;
+    }
```

Comment:
The switch statement is correctly used to handle different action types, which is good practice for a limited amount of conditions. However, remember to include a default case for error handling, even if it's just a break statement. This ensures that any unexpected action types are handled gracefully.